### ISSUE TITLE
`# noqa: RUF100` does not work

### ISSUE THREAD

**Issue by actionless** [charliermarsh/ruff#1302]
```
RUF100 Unused `noqa` directive for: RUF100
```

**Comment by charliermarsh** [comment]
Oh good call, thanks.

**Comment by charliermarsh** [comment]
@actionless - I have a fix in #1305, but could you post an example of how you expect this to work?

A few examples (`F401` is unused-import; `E501` is long-too-long):

```py
# This line doesn't trigger RUF100, so `noqa: RUF100` is an error
import os  # noqa: RUF100

---

# This line doesn't trigger RUF100, so `noqa: RUF100` is an error
import os  # noqa: F401, RUF100

---

# This line _would_ trigger RUF100, so `noqa: RUF100` suppresses it
import os  # noqa: E501, RUF100
```

**Comment by actionless** [comment]
in my case i just have a warning which i want to ignore in `flake`, but it's not raised by `ruff`, so it's raising RUF100

**Comment by charliermarsh** [comment]
Is it an error code that doesn't exist in Ruff? You could set `external = ["E301"]` or whatever in `pyproject.toml`.

**Comment by charliermarsh** [comment]
But, in that case, I should probably just do the simple thing and suppress `RUF100` whenever it's provided explicitly.

**Comment by actionless** [comment]
the situation even more complicated 😺
it exists in both flake and ruff, but behaves slightly differently (flake have a bug similar to https://github.com/charliermarsh/ruff/issues/1303 but only inside string types, ie `'Callable[[Any, DefaultNamedArg(bool | None, name="some_prop_name")], None]'` (note the quotes!)
so this will be failing the CI until it will start behaving same in both flake and ruff

**Comment by charliermarsh** [comment]
Hahaha ok!

**Comment by actionless** [comment]
after moving from noqa error codes to ruff-ignore with string rule names, the problem re-appearing again:

```
$ ruff --version
ruff 0.16.6
```

```
unused-noqa: [*] Unused suppression (unused: `unnecessary-if`, `unused-noqa`)
   --> pikaur_static/pacman_fallback.py:107:42
    |
105 |             line = await stream.readline()
106 |             if line:
107 |                 if self.enable_logging:  # ruff: ignore[unnecessary-if,unused-noqa]
    |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
108 |                     pass
109 |                 callback(line)
    |
help: Remove unused suppression
    |
106 |             if line:
    -                 if self.enable_logging:  # ruff: ignore[unnecessary-if,unused-noqa]
107 +                 if self.enable_logging:
108 |                     pass
    |
```

**Comment by ntBre** [comment]
Thanks, yeah this is tracked in https://github.com/astral-sh/ruff/issues/26282.
